unit/util/symbol_table.cpp

```diff
@@ -350,3 +350,28 @@ SCENARIO("journalling_symbol_table_writer",
     }
   }
 }
+
+TEST_CASE("symbol_tablet::lookup_ref invariant", "[core][utils][symbol_tablet]")
+{
+  symbolt foo_symbol;
+  foo_symbol.name = "foo";
+  symbol_tablet symbol_table;
+  symbol_table.insert(foo_symbol);
+  const cbmc_invariants_should_throwt invariants_throw;
+  REQUIRE_NOTHROW(symbol_table.lookup_ref("foo"));
+  try
+  {
+    symbol_table.lookup_ref("bar");
+    FAIL("Expected invariant failure.");
+  }
+  catch(const invariant_failedt &exception)
+  {
+    REQUIRE_THAT(
```

Comment:
Is this the best Catch can do for expected exceptions? No `REQUIRE_THROW` with `what` checking?

Reply:
I initially wrote these using `REQUIRE_THROWS_AS`, however that only allows checking the type of the exception, but not the `.what`. I think it could be done using `REQUIRE_THROWS_MATCHES` and a custom matcher. However I would have to spend a bit more time working out how to write such a matcher.

Reply:
I have a custom matcher working for this locally, but it should probably go in a follow up PR.